drivers/serial/uart_stm32.c

```diff
@@ -395,6 +395,46 @@ static void uart_stm32_poll_out(struct device *dev,
 	LL_USART_TransmitData8(UartInstance, (u8_t)c);
 }
 
+static int uart_stm32_err_check(struct device *dev)
+{
+	USART_TypeDef *UartInstance = UART_STRUCT(dev);
+	int err = 0;
+
+	if (LL_USART_IsActiveFlag_ORE(UartInstance)) {
+		err |= UART_ERROR_OVERRUN;
+	}
+
+	if (LL_USART_IsActiveFlag_PE(UartInstance)) {
+		err |= UART_ERROR_PARITY;
+	}
+
+	if (LL_USART_IsActiveFlag_FE(UartInstance)) {
+		err |= UART_ERROR_FRAMING;
+	}
+
+	if (err != 0) {
+#if defined(CONFIG_SOC_SERIES_STM32F4X) || \
```

Comment:
Please use LL API for this section. This will ease maintenance and introduction of new series.

Reply:
switched to LL, and added LPUART branch